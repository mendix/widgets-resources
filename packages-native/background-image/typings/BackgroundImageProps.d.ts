/**
 * This file was generated from BackgroundImage.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Team
 */
import { DynamicValue, NativeImage } from "mendix";
import { ReactNode } from "react";

interface CommonProps<Style> {
    name: string;
    style: Style[];
}

export type ImageResizeModeEnum = "cover" | "contain" | "stretch" | "repeat" | "center";

export interface BackgroundImageProps<Style> extends CommonProps<Style> {
    image: DynamicValue<NativeImage>;
    imageResizeMode: ImageResizeModeEnum;
    imageOpacity: BigJs.Big;
    content?: ReactNode;
}
