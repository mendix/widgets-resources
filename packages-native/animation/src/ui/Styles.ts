import { Style } from "@native-mobile-resources/util-widgets";
import { ViewStyle } from "react-native";

export interface AnimationStyle extends Style {
    container: ViewStyle;
}

export const defaultAnimationStyle: AnimationStyle = {
    container: {
        // All ViewStyle properties are allowed
    }
};
