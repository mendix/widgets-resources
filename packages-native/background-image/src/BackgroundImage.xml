<?xml version="1.0" encoding="utf-8" ?>
<widget id="com.mendix.widget.native.backgroundimage.BackgroundImage" supportedPlatform="Native" offlineCapable="true" pluginWidget="true" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../../../node_modules/mendix/custom_widget.xsd">
    <name>Background image</name>
    <description />
    <icon />
    <properties>
        <propertyGroup caption="General">
            <propertyGroup caption="General">
                <property key="image" type="image" required="true">
                    <caption>Image</caption>
                    <description />
                </property>
                <property key="imageResizeMode" type="enumeration" defaultValue="cover">
                    <caption>Image resize mode</caption>
                    <description>Determines how to resize the image when the widget dimensions don't match the raw image dimensions. For an SVG image it's always "stretch".</description>
                    <enumerationValues>
                        <enumerationValue key="cover">Cover</enumerationValue>
                        <enumerationValue key="contain">Contain</enumerationValue>
                        <enumerationValue key="stretch">Stretch</enumerationValue>
                        <enumerationValue key="repeat">Repeat</enumerationValue>
                        <enumerationValue key="center">Center</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="imageOpacity" type="decimal" defaultValue="1.0">
                    <caption>Image opacity</caption>
                    <description>Takes a value from 0.0 - 1.0. The lower the value, the more transparent.</description>
                </property>
                <property key="content" type="widgets" required="false">
                    <caption>Content</caption>
                    <description />
                </property>
            </propertyGroup>
            <propertyGroup caption="Common">
                <systemProperty key="Name" />
                <systemProperty key="Visibility" />
            </propertyGroup>
        </propertyGroup>
    </properties>
</widget>
