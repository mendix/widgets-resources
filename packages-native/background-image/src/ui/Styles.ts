import { ViewStyle, ImageStyle } from "react-native";
import { Style } from "@native-mobile-resources/util-widgets";

export interface BackgroundImageStyle extends Style {
    container: ViewStyle;
    image: ImageStyle;
}

export const defaultBackgroundImageStyle: BackgroundImageStyle = {
    container: {},
    image: {}
};
