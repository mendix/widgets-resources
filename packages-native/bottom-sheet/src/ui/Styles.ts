import { Style } from "@native-mobile-resources/util-widgets";
import { ViewStyle } from "react-native";

export interface BottomDrawerStyle extends Style {
    container: ViewStyle;
}

export const defaultBottomDrawerStyle: BottomDrawerStyle = {
    container: {
        flex: 1
    }
};
