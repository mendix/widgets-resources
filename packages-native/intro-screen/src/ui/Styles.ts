import { Style } from "@native-mobile-resources/util-widgets";
import { I18nManager, Platform, StyleSheet, TextStyle, ViewStyle } from "react-native";
import absoluteFillObject = StyleSheet.absoluteFillObject;
import DeviceInfo from "react-native-device-info";

const isiPhoneModelWithNotch = (): boolean => {
    const model = DeviceInfo.getDeviceId();
    if (model.indexOf("iPhone") !== -1) {
        switch (model) {
            case "iPhone10,6": // iPhone X GSM
            case "iPhone11,2": // iPhone XS
            case "iPhone11,4": // iPhone XS Max
            case "iPhone11,6": // iPhone XS Max Global
            case "iPhone11,8": // Iphone XR
            case "iPhone12,1": // Iphone 11
            case "iPhone12,3": // Iphone 11 Pro
            case "iPhone12,5": // Iphone 11 Pro Max
                return true;
            default:
                return false;
        }
    }
    return false;
};

const isAndroidRTL = I18nManager.isRTL && Platform.OS === "android";
const isIphoneWithNotch = DeviceInfo.hasNotch() || (Platform.OS === "ios" && isiPhoneModelWithNotch());

const defaultButtonBetweenContainer: ViewStyle = {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    paddingVertical: 12,
    alignSelf: "stretch"
};

const defaultButtonBelowContainer: ViewStyle = {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    paddingVertical: 12
};

const defaultButtonCaption: TextStyle = {
    backgroundColor: "transparent",
    color: "black",
    fontSize: 18,
    alignSelf: "center"
};

export interface ButtonStyle {
    container: ViewStyle;
    caption: TextStyle;
    icon: {
        color: string;
        size?: number;
    };
}

export interface IntroScreenStyle extends Style {
    fullscreenContainer: ViewStyle;
    popupContainer: ViewStyle;
    paginationContainer: ViewStyle;
    paginationText: TextStyle;
    dotStyle: ViewStyle;
    activeDotStyle: ViewStyle;
    paginationAbove: {
        buttonsContainer: ViewStyle;
        buttonSkip: ButtonStyle;
        buttonDone: ButtonStyle;
        buttonPrevious: ButtonStyle;
        buttonNext: ButtonStyle;
    };
    paginationBetween: {
        buttonSkip: ButtonStyle;
        buttonDone: ButtonStyle;
        buttonPrevious: ButtonStyle;
        buttonNext: ButtonStyle;
    };
}

export const defaultWelcomeScreenStyle: IntroScreenStyle = {
    fullscreenContainer: {
        ...absoluteFillObject
    },
    popupContainer: {
        ...absoluteFillObject,
        paddingHorizontal: 50,
        paddingVertical: 150,
        backgroundColor: "rgba(0, 0, 0, 0.5)"
    },
    paginationContainer: {
        position: "absolute",
        bottom: isIphoneWithNotch ? 22 : 0,
        left: 0,
        right: 0,
        justifyContent: "space-between",
        alignItems: "center"
    },
    paginationText: {
        fontSize: 12,
        color: "black"
    },
    paginationAbove: {
        buttonsContainer: {
            flex: 1,
            flexDirection: isAndroidRTL ? "row-reverse" : "row",
            justifyContent: "center",
            marginTop: 16
        },
        buttonSkip: {
            container: {
                ...defaultButtonBelowContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        },
        buttonDone: {
            container: {
                ...defaultButtonBelowContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        },
        buttonPrevious: {
            container: {
                ...defaultButtonBelowContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        },
        buttonNext: {
            container: {
                ...defaultButtonBelowContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        }
    },
    paginationBetween: {
        buttonSkip: {
            container: {
                ...defaultButtonBetweenContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        },
        buttonDone: {
            container: {
                ...defaultButtonBetweenContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        },
        buttonPrevious: {
            container: {
                ...defaultButtonBetweenContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        },
        buttonNext: {
            container: {
                ...defaultButtonBetweenContainer
            },
            caption: {
                ...defaultButtonCaption,
                paddingHorizontal: 5
            },
            icon: {
                color: "black"
            }
        }
    },
    dotStyle: {
        backgroundColor: "rgba(0, 0, 0, .2)"
    },
    activeDotStyle: {
        backgroundColor: "rgba(255, 255, 255, .9)"
    }
};
