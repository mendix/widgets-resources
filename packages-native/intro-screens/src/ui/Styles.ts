import { Style } from "@native-mobile-resources/util-widgets";
import { StyleSheet, TextStyle, ViewStyle } from "react-native";
import absoluteFillObject = StyleSheet.absoluteFillObject;

export interface IntroScreensStyle extends Style {
    fullscreenContainer: ViewStyle;
    cardContainer: ViewStyle;
    /**
     * Styles for bottom buttons
     */
    bottomButtonsContainer: ViewStyle;
    buttonSkip: ViewStyle;
    buttonDone: ViewStyle;
    buttonPrev: ViewStyle;
    buttonNext: ViewStyle;
    /**
     * Style for both texts
     */
    buttonText: TextStyle;
    /**
     * Styles for normal left/right button aligned to the pagination
     */
    leftButton: ViewStyle;
    rightButton: ViewStyle;
    /**
     * Styles for pagination
     */
    paginationContainer: ViewStyle;
    paginationDots: ViewStyle;
    dotStyle: ViewStyle;
    activeDotStyle: ViewStyle;
}

export const defaultWelcomeScreenStyle: IntroScreensStyle = {
    fullscreenContainer: {
        ...absoluteFillObject
    },
    cardContainer: {
        ...absoluteFillObject,
        paddingHorizontal: 50,
        paddingVertical: 100,
        backgroundColor: "rgba(0,0,0,0.4)"
    },
    bottomButtonsContainer: {},
    buttonSkip: {},
    buttonDone: {},
    buttonPrev: {},
    buttonNext: {},
    buttonText: {
        backgroundColor: "transparent",
        color: "white",
        fontSize: 18,
        padding: 12,
        alignSelf: "center"
    },
    leftButton: {
        position: "absolute",
        left: 0
    },
    rightButton: {
        position: "absolute",
        right: 0
    },
    paginationContainer: {},
    paginationDots: {},
    dotStyle: {
        backgroundColor: "rgba(0, 0, 0, .2)"
    },
    activeDotStyle: {
        backgroundColor: "rgba(255, 255, 255, .9)"
    }
};
