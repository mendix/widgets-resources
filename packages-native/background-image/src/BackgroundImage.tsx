import { createElement, ReactNode } from "react";
import { BackgroundImageStyle, defaultBackgroundImageStyle } from "./ui/Styles";
import { BackgroundImageProps } from "../typings/BackgroundImageProps";
import { ImageBackground } from "react-native";
import { ValueStatus } from "mendix";
import { flattenStyles } from "@native-mobile-resources/util-widgets";

export function BackgroundImage(props: BackgroundImageProps<BackgroundImageStyle>): ReactNode {
    const styles = flattenStyles(defaultBackgroundImageStyle, props.style);

    const backgroundImage =
        props.backgroundImage.status === ValueStatus.Available ? props.backgroundImage.value : undefined;

    if (!backgroundImage) {
        return null;
    }

    return (
        <ImageBackground source={backgroundImage} style={styles.container}>
            {props.content}
        </ImageBackground>
    );
}
