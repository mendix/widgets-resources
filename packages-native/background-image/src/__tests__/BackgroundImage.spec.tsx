import { createElement } from "react";
import { Text } from "react-native";
import { render } from "react-native-testing-library";
import { dynamicValue } from "@native-mobile-resources/util-widgets";

import { BackgroundImage } from "../BackgroundImage";
import { BackgroundImageProps } from "../../typings/BackgroundImageProps";
import { BackgroundImageStyle } from "../ui/Styles";

const defaultProps: BackgroundImageProps<BackgroundImageStyle> = {
    name: "backgroundImageTest",
    style: [],
    backgroundImage: dynamicValue({ uri: "path/to/image" }),
    content: <Text>Content</Text>
};

describe("BackgroundImage", () => {
    it("renders with default styles", () => {
        //@ts-ignore
        const component = render(<BackgroundImage {...defaultProps} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("renders with custom styles", () => {
        const style = [
            {
                container: { height: "50%" }
            },
            {
                container: { height: "80%" }
            }
        ];

        //@ts-ignore
        const component = render(<BackgroundImage {...defaultProps} style={style} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("renders nothing when image is unavailable", () => {
        const backgroundImage = dynamicValue();

        //@ts-ignore
        const component = render(<BackgroundImage {...defaultProps} backgroundImage={backgroundImage} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("renders wihtout content", () => {
        const content = null;

        //@ts-ignore
        const component = render(<BackgroundImage {...defaultProps} content={content} />);

        expect(component.toJSON()).toMatchSnapshot();
    });
});
