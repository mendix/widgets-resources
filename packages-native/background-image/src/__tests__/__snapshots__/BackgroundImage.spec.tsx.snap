// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage renders nothing when image is unavailable 1`] = `null`;

exports[`BackgroundImage renders wihtout content 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "path/to/image",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`BackgroundImage renders with custom styles 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "height": "80%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "path/to/image",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "80%",
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text>
    Content
  </Text>
</View>
`;

exports[`BackgroundImage renders with default styles 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "path/to/image",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text>
    Content
  </Text>
</View>
`;
